lib/inspec/reporters/base.rb

```diff
@@ -5,9 +5,36 @@ class Base
     def initialize(config)
       @config = config
       @run_data = config[:run_data]
+      apply_options_to_run_data
       @output = ""
     end
 
+    # Apply options such as message truncation and removal of backtraces
+    def apply_options_to_run_data
+      message_truncation = @config[:message_truncation] || "ALL"
+      trunc = -1
+      if message_truncation != "ALL"
+        if message_truncation.to_i.to_s != message_truncation
+          Inspec::Log.warn("Messages will not be truncated because #{message_truncation} is not an integer value")
```

Comment:
Consider making this an error - if they provided the option but it is unusable, halt and show usage of the bad option is a reasonable thing to do. Continuing and generating too much data might be worse.

Reply:
My thinking for making this is a warning is that:
 * The CLI, not this class, is responsible for config validation (even though it's not doing any currently...). 
 * At this point, the run has completed. It possibly took a long time and burned through billable compute resource. The reporter should generally never halt.

Possibly I could make the option strictly numeric and throw away the `ALL` special case - I did it this way so that those using InSpec in CI could have a variable driven run command where `message-truncation` is always set at the CLI and the value is defined by a CI variable.

Reply:
Good point about the validation happening post-run - perhaps the validation should be moved to be here - https://github.com/inspec/inspec/blob/2e887a94afcca819da781d4774aa2a5a0b56785e/lib/inspec/config.rb#L328

Reply:
Thanks, that looks like the right spot for validation.
This warning was meant to supplementary to proper validation - but I think if there's validation in `config.rb` then this class also doing a check is overkill - so I'll just remove this warning.